napari/utils/translations.py

```diff
@@ -0,0 +1,456 @@
+"""
+Localization utilities to find available language packs and packages with
+localization data.
+"""
+
+import gettext
+import importlib
+import json
+import os
+import subprocess
+import sys
+import traceback
+from pathlib import Path
+
+import pkg_resources
+
+# Entry points
+NAPARI_LANGUAGEPACK_ENTRY = "napari.languagepack"
+
+# Constants
+DEFAULT_LOCALE = "en"
+LOCALE_DIR = "locale"
+
+
+def _main():
+    """
+    Run functions in this file in a subprocess and prints to stdout the results.
+    """
+    data = {}
+    message = ""
+    if len(sys.argv) == 2:
+        func_name = sys.argv[-1]
+        func = globals().get(func_name, None)
+
+        if func:
+            try:
+                data, message = func()
+            except Exception:
+                message = traceback.format_exc()
+    else:
+        message = "Invalid number of arguments!"
+
+    sys.stdout.write(json.dumps({"data": data, "message": message}))
+
+
+def _get_installed_language_pack_locales() -> dict:
+    """
+    Get available installed language pack locales.
+
+    Returns
+    -------
+    dict
+        A dict with all the languages found.
+
+    Notes
+    -----
+    This functions are meant to be called via a subprocess to guarantee the
+    results represent the most up-to-date entry point information, which
+    is defined on interpreter startup.
+    """
+    data = {}
+    messages = []
+    for entry_point in pkg_resources.iter_entry_points(
+        NAPARI_LANGUAGEPACK_ENTRY
+    ):
+        try:
+            data[entry_point.name] = os.path.basename(
+                os.path.dirname(entry_point.load().__file__)
+            )
+        except Exception:
+            messages.append(traceback.format_exc())
+
+    message = "\n".join(messages)
+    return data, message
+
+
+def _get_display_name(
+    locale: str, display_locale: str = DEFAULT_LOCALE
+) -> str:
+    """
+    Return the language name to use with a `display_locale` for a given language locale.
+    Parameters
```

Comment:
I think we generally have empty lines above `Parameters` and `Returns`

Reply:
is this function used anywhere besides the tests?  If not, can it just move to the tests file?

Reply:
This will be used to display the available languages in the preferences.

Reply:
Fixed. I forgot to add a method. See bleow.